yarn-project/simulator/src/avm/journal/journal.ts

```diff
@@ -170,12 +172,18 @@ export class AvmPersistableStateManager {
       const lowLeafPath = lowLeafInfo.siblingPath;
 
       const newLeafPreimage = result.element as PublicDataTreeLeafPreimage;
-      let insertionPath;
-
+      let insertionPath: Fr[] | undefined;
       if (!result.update) {
         insertionPath = result.insertionPath;
       }
 
+      if (lowLeafIndex !== 0n) {
```

Comment:
I'm not sure I understand this `if` condition, there might be (unlikely) situation where the `lowLeafIndex === 0`.

If we are testing a write, we should re-use the `!result.update` condition from above